homeassistant/components/qbittorrent/sensor.py

```diff
@@ -1,68 +1,65 @@
 """Support for monitoring the qBittorrent API."""
 import logging
 
-from qbittorrent.client import Client, LoginRequired
+from qbittorrent.client import LoginRequired
 from requests.exceptions import RequestException
-import voluptuous as vol
-
-from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import (
-    CONF_NAME,
-    CONF_PASSWORD,
-    CONF_URL,
-    CONF_USERNAME,
-    DATA_RATE_KILOBYTES_PER_SECOND,
-    STATE_IDLE,
-)
-from homeassistant.exceptions import PlatformNotReady
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.entity import Entity
 
-_LOGGER = logging.getLogger(__name__)
+from homeassistant.const import CONF_URL, DATA_RATE_KILOBYTES_PER_SECOND, STATE_IDLE
+from homeassistant.helpers.entity import Entity
 
-SENSOR_TYPE_CURRENT_STATUS = "current_status"
-SENSOR_TYPE_DOWNLOAD_SPEED = "download_speed"
-SENSOR_TYPE_UPLOAD_SPEED = "upload_speed"
+from .const import (
+    DATA_KEY_CLIENT,
+    DATA_KEY_NAME,
+    DOMAIN,
+    SENSOR_TYPE_ACTIVE_TORRENTS,
+    SENSOR_TYPE_COMPLETED_TORRENTS,
+    SENSOR_TYPE_CURRENT_STATUS,
+    SENSOR_TYPE_DOWNLOAD_SPEED,
+    SENSOR_TYPE_DOWNLOADING_TORRENTS,
+    SENSOR_TYPE_INACTIVE_TORRENTS,
+    SENSOR_TYPE_PAUSED_TORRENTS,
+    SENSOR_TYPE_RESUMED_TORRENTS,
+    SENSOR_TYPE_SEEDING_TORRENTS,
+    SENSOR_TYPE_TOTAL_TORRENTS,
+    SENSOR_TYPE_UPLOAD_SPEED,
+    TRIM_SIZE,
+)
+from .wrapper_functions import get_main_data_client, retrieve_torrentdata
 
-DEFAULT_NAME = "qBittorrent"
+_LOGGER = logging.getLogger(__name__)
 
 SENSOR_TYPES = {
     SENSOR_TYPE_CURRENT_STATUS: ["Status", None],
     SENSOR_TYPE_DOWNLOAD_SPEED: ["Down Speed", DATA_RATE_KILOBYTES_PER_SECOND],
     SENSOR_TYPE_UPLOAD_SPEED: ["Up Speed", DATA_RATE_KILOBYTES_PER_SECOND],
+    SENSOR_TYPE_TOTAL_TORRENTS: ["Total Torrents", None],
+    SENSOR_TYPE_ACTIVE_TORRENTS: ["Active Torrents", None],
+    SENSOR_TYPE_INACTIVE_TORRENTS: ["Inactive Torrents", None],
+    SENSOR_TYPE_DOWNLOADING_TORRENTS: ["Downloading", None],
+    SENSOR_TYPE_SEEDING_TORRENTS: ["Seeding", None],
+    SENSOR_TYPE_RESUMED_TORRENTS: ["Resumed Torrents", None],
+    SENSOR_TYPE_PAUSED_TORRENTS: ["Paused Torrents", None],
+    SENSOR_TYPE_COMPLETED_TORRENTS: ["Completed Torrents", None],
 }
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_URL): cv.url,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    }
-)
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
We can import the yaml config to a config flow to make the transition smoother for users.

Reply:
good suggestion, this way we I can remove the breaking change tag

Reply:
Added to the sensor again